app/assets/javascripts/welcome.coffee

```diff
@@ -35,3 +35,17 @@ $ ->
       success: (data) ->
         $('#partial_news').html data
     event.preventDefault()
+
+  $('#myCarousel').carousel()
+  item = $('.carousel .item')
+  item.eq(0).addClass('active')
+  numberofSlides = $('.item').length
+  currentSlide = Math.floor(Math.random() * numberofSlides)
+  $('.carousel-indicators li').each ->
```

Comment:
if it is used only here and not big, just create this method here, if it is repeated on few places than you need to separate it